apis/v1alpha1/shared_types.go

```diff
@@ -182,6 +182,7 @@ type RouteGatewayStatus struct {
 	// +listType=map
 	// +listMapKey=type
 	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:default={{type: "Admitted", status: "False", message: "Waiting for controller", reason: "Waiting", lastTransitionTime: "1970-01-01T00:00:00Z"}}
```

Comment:
Wouldn't this RouteGatewayStatus object only exist if it was added by a controller to begin with?

Reply:
That sounds right. Do we have a bug in the API then? There should be a way to reflect this default status.

Reply:
This specific object only gets added when a Route binds to a Gateway. That is logic controllers will be responsible for. I think the current empty default state here for the parent list is correct - no Gateways are using this Route. When that changes, it will because a controller is connecting the two, and at that point it becomes the controller's responsibility to set this status for the Gateway(s) it is managing. On that note, the associated godoc here could really use some more detail around what we're expecting of controllers here.

Reply:
My last comment was ambiguous. What I mean is, when a Route is created and no controller has admitted it, should there not be a default condition on the route which communicates the same to the user?
We can educate the users to think of a Route without any conditions equivalent to a Route not yet admitted. I think that might trip some users.

Reply:
If I'm understanding the structure correctly, the current default status for all routes would be:

```yaml
status:
  gateways: []
```

That seems like a reasonable starting point. Although it's not completely obvious, I think that many will be able to interpret that as "no Gateways are using this". 

I think the only way we could make this more clear would be to add a separate "conditions" list that could have a default value that indicated the route was not in use. Unfortunately with any conditions applied to the route as a whole it becomes rather difficult to determine which controller is responsible for updating them, since 0-multiple controllers may be processing the same route at any moment in time. 

If we did add that default condition, we could state that the first controller to add a gateway to status would be responsible for removing the condition. It becomes messier on removal though, especially if multiple controllers are removing Gateways at the same time.

Reply:
Hmm, now I recall that we had a similar discussion before. Maybe we should explicitly document that no status means the route has not been admitted. Far from perfect, but it is something.

Reply:
Agree with that, I think that's probably the best option we have here. What do you think @danehans?

Reply:
Swinging back in after having been away for a while, sorry!

For Conditions, it's really supposed to be that the default value is "Unknown", where that means "no controller has claimed this yet".

In this case, I think that we should both document that the empty slice means "not owned by anything", but default the Conditions slice to have one entry of `type: Admitted`, `status: Unknown`, with the current Reason of `To be updated by the owning controller` or something similar? That way, once a controller does add a RouteGatewayStatus, the `Admitted` condition will be there by default.

Reply:
Welcome back! This is an interesting situation because it seems like the only entity that would add a Gateway to this list would be the owning controller. So a default would only apply when that controller was adding the Gateway to the list, and hopefully also setting a value inside conditions. If the controller did not set any value in conditions, is it more confusing for the slice to be empty or to have a "To be updated by the owning controller" condition despite actually being set by the owning controller? Neither option is great, but I'm not sure a default value here is very helpful. Maybe we need to provide clearer guidance that a condition must be set here? We could even add validation that ensured at least 1 condition was set when a Gateway was added here.

Reply:
> Agree with that, I think that's probably the best option we have here. What do you think @danehans?

I think a reasonable next step is to update the godocs to explain that an empty `status.gateways` means the route has not been admitted by any gateway.